### DISCUSSION TITLE
q-input problem after upgrade to v2

### DISCUSSION THREAD

**Discussion by ssuess** [quasarframework/quasar#13814] [General - Components / Directives / etc]
I have a problem upgrading a part of my application to Quasar v2. I have a search input that triggers an api call that populates a dropdown when values are entered in the field. Here is my input:
```
<q-input filled type="search" inputmode="search" v-on:keyup.enter="commitSearch(searchInput, false);ACreturnoptions = [];" debounce="400" v-on:keyup.down="focusmenu" @update:model-value="filterFn(searchInput);menuenter=false;showing=true;" @click="showing=true;menuenter=false;" v-model.lazy="searchInput" square :label="$t('searchPlaceholder')" class='sinputbar bg-sb'  ref="searchfield">
```
As suggested by the migration docs, I changed  my v1 version from`@input="filterFn(searchInput);menuenter=false;showing=true;"` to `@update:model-value="filterFn(searchInput);menuenter=false;showing=true;"`

in v1, this worked as expected. In v2:
- filterFn only seems to tigger with the PREVIOUS value entered in the search field, not the current one
- in vue devtools I can see that `searchInput` IS updating in the data object, but
- I can workaround these problems by using a watch on `searchInput`, but I am confused as to why `@update:model-value` is not working as expected

Also, trying to change `v-model` to `:model-value` failed, it would keep clearing the field and the value would not update at all.

**Comment by suleimanas** [comment]
try it with only one function

```
<q-input ... @update:model-value="searchUpdate" />
function searchUpdate(searchInput){
  // filterFn(searchInput);menuenter=false;showing=true;
}
```

**Reply by ssuess** [reply]
I'll try that thanks, but not sure why it would have worked fine in v1...
